config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -74,11 +75,28 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Bundle
         }
     }
 
-    public BundleMetadata buildEnvironmentMetadata(final List<Entity> entities, ProjectInfo projectInfo){
-        BundleMetadata.Builder builder = new BundleMetadata.Builder(EntityBuilder.BundleType.ENVIRONMENT.name(), projectInfo.getName(),
-                projectInfo.getName(), projectInfo.getGroupName(), projectInfo.getVersion());
+    /**
+     * Generates metadata for environment bundle
+     * @param entities List of environment entities
+     * @param projectInfo project information configured in build.gradle
+     * @return BundleMetaData
+     */
+    public BundleMetadata buildEnvironmentMetadata(final List<Entity> entities, ProjectInfo projectInfo) {
+        final boolean isConfigNamePresent = StringUtils.isNotBlank(projectInfo.getConfigName());
+        String version = projectInfo.getVersion();
+        if (StringUtils.isNotBlank(version) && isConfigNamePresent) {
+            version = version + "-" + projectInfo.getConfigName();
+        }
```

Comment:
So if `version` is blank, we will skip the `configName` as well?

Reply:
yes